File: PyDS/selection_sort.py
def selectionSort(lyst):
    n = len(lyst)
    for i in range(0, n - 1):
        minIndex = i
        for j in range(i + 1, n):
            if lyst[j] < lyst[minIndex]:
                minIndex = j

        # i如果已经在正确的位置上，则不需要交换
        if minIndex != i:
            swap(lyst, minIndex, i)


def swap(lyst, i, j):
    temp = lyst[i]
    lyst[i] = lyst[j]
    lyst[j] = temp


lyst = [4, 5, 6, 1, 2, 3]

selectionSort(lyst)
print(lyst)

File: Others/JS/JSDS/selection_sort.js
function selectionSort(array) {
    for (var i = 0; i < array.length - 1; i++) {
        var minIndex = i
        for (var j = i + 1; j < array.length; j++) {
            if (array[j] < array[minIndex]) {
                minIndex = j
            }
        }

        if (minIndex != i) {
            swap(array, minIndex, i)
        }
    }
}

function swap(array, i, j) {
    var temp = array[i]
    array[i] = array[j]
    array[j] = temp
}

array = [4, 5, 6, 1, 2, 3]
selectionSort(array)
console.log(array)